app.js

```diff
@@ -38,29 +37,16 @@ app.listen(port, () => {
   console.log('Listening on port ' + port + '   url: http://localhost:' + port)
 })
 
-// Create a componentDirectory object representing the components directory
-// which we can use to iterate the component list on the homepage
-dto(path.resolve('./src/components'), (err, res) => {
-  if (err) {
-    console.log(err)
-  }
-  app.locals.componentsDirectory = res
-})
-
-// Create an examplesDirectory object representing the examples directory
-// which we can use to iterate the list of examples on the examples page
-dto(path.resolve('./src/views/examples'), (err, res) => {
-  if (err) {
-    console.log(err)
-  }
-  app.locals.examplesDirectory = res
-})
-
 // Define routes
 
 // Index page - render the component list template
 app.get('/', function (req, res) {
-  res.render('component-list')
+  // Creates a componentsDirectory object representing the components in the src directory
+  // which is then passed to the view to list the components
+  helperFunctions.directoryToObject(path.resolve('./src/components'))
```

Comment:
My understanding is that the way the code previously worked was to define the directory object once when the app was started, and share that information between routes.

If I'm understanding this correctly, the directory object will now be regenerated every time the route is requested. Is that intentional, and do we understand the consequences of making this change? If generating the directory object is particularly intensive or time-consuming do we want to do this?

Reply:
```define the directory object once when the app was started, and share that information between routes.```
Yes that is right. There is a change that I haven't captured in my PR comment and that should be up for discussion.

My expectation is to have a list of directories that reflects the directory structure at any moment. So if I add/remove a component after the server is started I'd like the listing to reflect that, because otherwise, for example, I might click on a link for a component that no longer exists. Indeed that means reading the directory each time I access the page.

If we find that's too intensive/time-consuming and decide to read the directories only when starting the server, then I can update the code to do that.

The main changes here are actually about:
 - making that data available to specific pages
 - wait for the promise to be resolved before displaying a page

Reply:
another way of approaching updating is to use something like nodemon in dev mode to restart everything when changes happen - that's the way the Prototype Kit works

Reply:
Happy with all of the above, just want to check we've thought it through.

We can always revisit if it becomes problematic.